File: Src/Icm.Core/CommandLine/UnnamedParameter.vb

Namespace Icm.CommandLineTools
    ''' <summary>
    '''  Unnamed parameter.
    ''' </summary>
    ''' <remarks></remarks>
    Public Class UnnamedParameter

        Private ReadOnly _description As String
        Private ReadOnly _name As String

        Public Sub New( _
                ByVal name As String,
                ByVal description As String)
            _name = name
            _description = description
        End Sub

        ''' <summary>
        '''  Long option (for example --help)
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        ReadOnly Property Name() As String
            Get
                Return _name
            End Get
        End Property

        ''' <summary>
        ''' Description of this option.
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        ReadOnly Property Description() As String
            Get
                Return _description
            End Get
        End Property

    End Class

End Namespace

File: Src/Icm.Core/CommandLine/UnnamedParameter.cs
namespace Icm.CommandLineTools
{
	/// <summary>
	///  Unnamed parameter.
	/// </summary>
	/// <remarks></remarks>
	public class UnnamedParameter
	{

		private readonly string _description;

		private readonly string _name;
		public UnnamedParameter(string name, string description)
		{
			_name = name;
			_description = description;
		}

		/// <summary>
		///  Long option (for example --help)
		/// </summary>
		/// <value></value>
		/// <returns></returns>
		/// <remarks></remarks>
		public string Name {
			get { return _name; }
		}

		/// <summary>
		/// Description of this option.
		/// </summary>
		/// <value></value>
		/// <returns></returns>
		/// <remarks></remarks>
		public string Description {
			get { return _description; }
		}

	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
